src/pykeen/datasets/base.py

```diff
@@ -223,22 +225,89 @@ def __repr__(self) -> str:  # noqa: D105
         )
 
 
-def _urlretrieve(url, path, clean_on_failure: bool = True) -> None:
+def _name_from_url(url: str) -> str:
+    """Get the filename form the end of the URL."""
+    parse_result = urlparse(url)
+    return os.path.basename(parse_result.path)
```

Comment:
Does this work on Windows, were the line separator is `\` ? Also, using a path utility to break URI into parts... :thinking: 

`parse_result.path.rsplit("/", maxsplit=1)[-1]` should be safe.

Reply:
You're right... there must be a better way though

Reply:
Resolved in bc80c729 with pathlib